src/mlpack/tests/main_tests/adaboost_test.cpp

```diff
@@ -0,0 +1,267 @@
+
+  trainData.shed_row(trainData.n_rows - 1);
+
+  // Now train Adaboost with labels provided.
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("test", std::move(testData));
+  SetInputParam("labels", std::move(labels));
+
+  mlpackMain();
+
+  // Check that initial output and final output matrix are same.
+  CheckMatrices(output, CLI::GetParam<arma::Row<size_t>>("output"));
+}
+
+/**
+ * Testing that only one of training data or pre-trained model is passed.
+ */
+BOOST_AUTO_TEST_CASE(AdaBoostTrainingDataOrModelTest)
+{
+  arma::mat trainData;
+  if (!data::Load("trainSet.csv", trainData))
+    BOOST_FAIL("Unable to load train dataset trainSet.csv!");
+
+  SetInputParam("training", std::move(trainData));
+
+  mlpackMain();
+
+  SetInputParam("input_model",
+                std::move(CLI::GetParam<AdaBoostModel>("output_model")));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Weak learner should be either Decision Stump or Perceptron.
+ */
+
+BOOST_AUTO_TEST_CASE(AdaBoostWeakLearnerTest)
+{
+  arma::mat trainData;
+  if (!data::Load("trainSet.csv", trainData))
+    BOOST_FAIL("Unable to load train dataset trainSet.csv!");
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("weak_learner", std::string("decision tree"));
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Weak learner should be ignored if it is
+ * specified with an input model file.
+ */
+BOOST_AUTO_TEST_CASE(AdaBoostWeakLearnerIgnoredTest)
+{
+  arma::mat trainData;
+  if (!data::Load("vc2.csv", trainData))
+    BOOST_FAIL("Unable to load train dataset vc2.csv!");
+
+  arma::Row<size_t> labels;
+  if (!data::Load("vc2_labels.txt", labels))
+    BOOST_FAIL("Unable to load label dataset vc2_labels.txt!");
+
+  arma::mat testData;
+  if (!data::Load("vc2_test.csv", testData))
+    BOOST_FAIL("Unable to load test dataset vc2.csv!");
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(labels));
+
+  SetInputParam("test", testData);
+
+  mlpackMain();
+
+  CLI::GetSingleton().Parameters()["training"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["test"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["labels"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["weak_learner"].wasPassed = false;
+
+  // Default value is Decision Stump
+  SetInputParam("input_model",
+                std::move(CLI::GetParam<AdaBoostModel>("output_model")));
+  SetInputParam("test", std::move(testData));
+  SetInputParam("weak_learner", std::string("perceptron"));
+
+  mlpackMain();
+
+  const string weakLearner = CLI::GetParam<string>("weak_learner");
+  if (weakLearner == "perceptron")
+  {
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+  }
```

Comment:
I don't understand what you are trying to test here, could you clarify please?

Reply:
If a model is given as input, and after that a different weak learner is specified (Maybe with the testing data), it should be ignored.
The same is written in adaboost_main.cpp line 132. I'm trying to test this. 
Is it wrong or have I misunderstood it in some way?

Reply:
But if the weak learner is ignored, the program won't throw an exception (or at least it shouldn't), it should just print a warning message.  I don't think we have a good way to test if warning output is given, so I wouldn't worry about testing that.